cmd/yggd/main.go

```diff
@@ -166,25 +176,25 @@ func main() {
 			return cli.Exit(fmt.Errorf("cannot kill workers: %w", err), 1)
 		}
 
-		clientIDFile := filepath.Join(yggdrasil.LocalstateDir, yggdrasil.LongName, "client-id")
-		if c.String("cert-file") != "" {
-			CN, err := parseCertCN(c.String("cert-file"))
+		clientIDFile := filepath.Join(yggdrasil.LocalstateDir, yggdrasil.LongName, cliClientID)
+		if c.String(cliCertFile) != "" {
```

Comment:
Again, if we're going to move to a global `main.Config` struct, this kind of initialization should probably be done from the values pulled out of the config variable. I can see some initialization code that puts together the config variable values from the `cli.Context`, while using the values from the config variable to further assemble the rest of the config variable.

```go
config = Config{}
config.ClientId = c.String("client-id")
config.CertFile = c.String("cert-file")

clientIDFile := filepath.Join(yggdrasil.LocalstateDir, yggdrasil.LongName, config.ClientID)
if config.CertFile != "" {
```